GooseSLURM/cli/Gsub.py

```diff
@@ -77,68 +84,119 @@ def dump(files, ifile, outname):
+    parser.add_argument("-w", "--wait", type=float, default=0.1)
+    parser.add_argument("-s", "--serial", action="store_true")
+    parser.add_argument("--dry-run", action="store_true")
+    parser.add_argument("--verbose", action="store_true")
+    parser.add_argument("-q", "--quiet", action="store_true")
+    parser.add_argument("-v", "--version", action="version", version=__version__)
+    parser.add_argument("files", nargs="*", type=str)
+    args = parser.parse_args()
 
-    # parse command-line options
-    args = docopt.docopt(__doc__, version=__version__)
-    files = args['<files>']
+    if not args.constrain:
+        args.constrain = [None]
+
+    if args.dry_run:
+        args.quiet = True
+        args.verbose = True
 
     # checkout existing output
-    if args['--output']:
-        if not fileio.ContinueDump(args['--output']):
+    if args.output:
+        if not fileio.ContinueDump(args.output):
             return 1
 
-    # read YAML-file
-    if args['--input']:
-        try:
-            source = args['--input']
-            key = list(filter(None, args['--key'].split('/')))
-            files = fileio.YamlGetItem(source, key)
-        except Exception as e:
-            print(e)
-            return 1
+    # interpret YAML-file
+    if args.input:
+        assert os.path.isfile(os.path.realpath(args.input))
+        assert len(args.files) == 0
+        key = list(filter(None, args["--key"].split("/")))
+        args.files = fileio.YamlGetItem(args.input, key)
 
-    # check arguments
-    for file in files:
-        if not os.path.isfile(file):
-            print('"%s" does not exist' % file)
-            return 1
+    assert len(args.files) > 0
+    assert all([os.path.isfile(os.path.realpath(file)) for file in args.files])
+
+    # construct command
+    sbatch = ["sbatch"]
+    for key, opt in zip(["time", "account"], [args.time, args.account]):
+        if opt:
+            sbatch += [f"--{key} {opt}"]
 
     # submit
-    pbar = tqdm.tqdm(files, disable=args['--quiet'])
-
-    for ifile, file in enumerate(pbar):
-        pbar.set_description(file)
-        path, name = os.path.split(file)
-
-        if len(path) > 0:
-            cmd = 'cd {0:s}; sbatch {1:s}'.format(path, name)
-        else:
-            cmd = 'sbatch {0:s}'.format(name)
-
-        out = run(cmd, verbose=args['--verbose'], dry_run=args['--dry-run'])
-        dump(files, ifile + 1, args['--output'])
-
-        if args['--serial']:
-            jobid = out.split('Submitted batch job ')[1]
-            time.sleep(20)
-            while True:
-                start = time.time()
-                status = run('squeue -j {0:s}'.format(jobid)).split('\n')
-                end = time.time()
-                if len(status) == 1:
-                    break
-                if len(status) == 2:
-                    if len(status[1]) == 0:
+
+    cbar = tqdm.tqdm(args.constrain, disable=args.quiet or len(args.constrain) == 1)
+
+    for constrain in cbar:
```

Comment:
For this to be meaningful there should be a loop on output files as well 
```suggestion
    for constrain, output in zip(cbar, ...):
```